esmvaltool/recipes/recipe_quantilebias.yml

```diff
@@ -32,20 +34,20 @@ diagnostics:
     description: Tool for calculation of precipitation quantile bias 
     variables:
       pr:
-        preprocessor: masking_preprocessor 
+        preprocessor: mask_regrid_preproc 
         reference_dataset: "GPCP-SG"
         mip: Amon
         project: CMIP5
         exp: historical
         ensemble: r1i1p1
-        start_year: 1997
-        end_year: 1997
+        start_year: 1979
+        end_year: 2005
     additional_datasets:
-      - {dataset: ACCESS1-0}
-      - {dataset: ACCESS1-3}
+      #- {dataset: ACCESS1-0}
```

Comment:
Why does this recipe contain so many commented out datasets?

Reply:
Good point. I cleaned it up now. I actually reverted the logic, so now the GPCP observations are specified as a "additional_dataset" for the pr variable, while the model dataset is among datasets. All  other recipes by the way seem to be different in their approach to this: sometimes the reference dataset is specified among general datasets, sometimes as additional_datasets, sometimes it is the opposite .... is there some logic/guideline on this?
In this case it probably makes sense to specify the GPCP obs4mip dataset close to the variable and to specify separately "mip: Amon" for the variable, since the diagnostic really should be used only for monthly data. But for the rest I am a bit confused by the too many possibilities to obtain the same result :)

Reply:
> is there some logic/guideline on this?

In general I would recommend defining all datasets in one place, to keep it simple. However, in some cases a dataset is only available for some of the variables in the recipe or diagnostic, so in that case you need the `additional_datasets` keyword at variable level. We actually have an open issue about simplifying this: https://github.com/ESMValGroup/ESMValCore/issues/92, feel free to comment if you have an opinion ;-)